src/CTTConfig.py

```diff
@@ -0,0 +1,157 @@
+                            help='Use the board configuration mail recipients')
+        parser.add_argument('--notify', nargs='+',
+                            help='Mail recipients of the notifications')
+        parser.add_argument('-b', '--boards', nargs='+',
+                            help='Board to run the test on')
+        parser.add_argument('-l', '--list', action='store_true',
+                            help='List all the available boards')
+
+        job = parser.add_argument_group('Job handling')
+        job.add_argument('--output-dir', default='jobs',
+                         help='Path where the jobs will be stored')
+        job.add_argument('--rootfs-path',
+                         help='Path to the prebuilt rootfs images directory')
+        job.add_argument('--job-name', help='Name of the job')
+
+        job.add_argument('--rootfs', help='Path to your rootfs image')
+        job.add_argument('--kernel', help='Path to your kernel image')
+        job.add_argument('--dtb', help='Path to your dtb')
+        job.add_argument('--dtb-folder', help='Path to your dtb folder')
+        job.add_argument('--modules', help='Path to your modules tar.gz')
+        job.add_argument('-t', '--tests', nargs='+',
+                         help='Tests to run on the board')
+
+        lava = parser.add_argument_group('LAVA server options')
+        lava.add_argument('--server', help='LAVA server to send results to')
+        lava.add_argument('--username', help='LAVA username')
+        lava.add_argument('--token', help='LAVA token')
+
+        artifacts = parser.add_argument_group('Artifacts options')
+        artifacts.add_argument('--api-token', help='KernelCI API token')
+        artifacts.add_argument('--tree', default='mainline',
+                               help='KernelCI tree to use')
+        artifacts.add_argument('--branch', default='master',
+                               help='KernelCI branch to use')
+        artifacts.add_argument('--defconfigs', nargs='+',
+                               help='KernelCI defconfig to use')
+        artifacts.add_argument('--no-kci', action='store_true',
+                               help='Ignore KernelCI artifacts')
+
+        ssh = parser.add_argument_group('SSH server options')
+        ssh.add_argument('--ssh-server', help='SSH server')
+        ssh.add_argument('--ssh-username', help='SSH username')
+
+        self.__cmdline = vars(parser.parse_args())
+
+    def __validate_config_file(self):
+        if not self.__config.has_section(DEFAULT_SECTION):
+            raise SectionError('Missing %s section' % DEFAULT_SECTION)
+
+    def __validate_cmdline(self):
+        # We can always just list the boards
+        if self.__cmdline['list']:
+            return
+
+        # Validate that we have all the basic options...
+        for option in mandatory_keys:
+            if not self.__contains__(option):
+                raise OptionError('Missing %s option' % option)
```

Comment:
Hmmm... I just think we should really split the options of the config file and the options for command line. Maybe this would help making the options less messy.